src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc

```diff
@@ -22,6 +22,11 @@
     #?(:clj [cljs.analyzer :as ana])
     [com.fulcrologic.fulcro.algorithms.indexing :as indexing]))
 
+;; ident of the current component, derived from its class.
+;; Since we don't know its props yet, if the "id" part depends on props,
+;; it will be `nil`.
+(def ^:dynamic *current-comp-ident* nil)
```

Comment:
I'm not loving the dynamic var. We're exposing something for convenience here that isn't terribly useful for anything but this error check. It might make sense to instead expose something that's a bit more useful, such as the class itself? The thing is I don't want something globally visible like this that isn't generally useful, and since we also don't have a clear (documented) story of when that will be valid it could be quite confusing, since it is only valid in very special circumstances.

Reply:
Ok, I have changed that to `*target-class* and set it now for each of the three notification functions. Is that better? If not, do you have any tip how to solve it differently? How else could route-immediate/... get the info inside which component it is invoked?